### PR TITLE
Remove channel publishing and clean up PublishData

### PR DESCRIPTION
**Pull request by JoeRobich**
Remove references to non-servicing branches. Remove channel data to stop updating dotnet/versions repo.

### CODE REVIEW

File: eng/config/PublishData.json [hunk 1]
```diff
@@ -121,120 +121,57 @@
       "version": "2.10.*",
       "nuget": [ ],
       "vsix": null,
-      "channels": [ "dev15.9" ],
+      "channels": [ ],
       "vsBranch": "rel/d15.9",
       "vsMajorVersion": 15
     },
-    "dev16.0-vs-deps": {
```
Comment (RikkiGibson) [inline eng/config/PublishData.json:128]: Should we also link to the servicing doc like we did here:

https://github.com/dotnet/roslyn-tools/blob/9528c5b1c7c625f87b55ec9e805750e4b9b3d7b6/src/GitHubCreateMergePRs/config.xml#L3

**Review by jaredpar — APPROVED** [review]

File: eng/config/PublishData.json (hunk 1 above)
Comment (JoeRobich) [reply eng/config/PublishData.json:128]: sicne this isn't jsonc that is a little more complicated, but maybe I can add a "comment" property somewhere.

Comment (RikkiGibson) [reply eng/config/PublishData.json:128]: Perhaps a comment at the point where the PublishData is parsed in the bulid script?

**Review by RikkiGibson — APPROVED** [review]

File: eng/config/PublishData.json (hunk 1 above)
Comment (RikkiGibson) [reply eng/config/PublishData.json:128]: Or is it parsed by an external service? I don't remember.

Comment (JoeRobich) [reply eng/config/PublishData.json:128]: Added a comment in the publishdata.json.

File: eng/config/PublishData.json [hunk 2]
```diff
@@ -115,126 +115,64 @@
       "Microsoft.VisualStudio.LanguageServices.Razor.RemoteClient": "arcade"
     }
   },
+  "comment-about-servicing-branches": "For a list of VS versions under servicing, see https://docs.microsoft.com/en-us/visualstudio/releases/2019/servicing#support-options-for-enterprise-and-professional-customers",
```
Comment (RikkiGibson) [inline eng/config/PublishData.json:118]: works for me, thanks!

**Pull request merged**